tcms/testcases/migrations/0011_trim_bugsystem_fields.py

```diff
@@ -1,6 +1,37 @@
 # Generated by Django 2.2.4 on 2019-08-16 12:17
 
-from django.db import migrations
+from django.db import migrations, models
+
+
+def forwards_store_data(apps, schema_editor):
+    bug_system_model = apps.get_model('testcases', 'BugSystem')
+    with open('/tmp/bugsystems-migration-dump-0011-\
+trim-bugsystem-fields',  # nosec:B108:hardcoded_tmp_directory
+              'w') as temp_file:
+        for bug_system in bug_system_model.objects.all():
+            temp_file.write('{},{},{},{}\n'
+                            .format(bug_system.name,
+                                    bug_system.description,
+                                    bug_system.url_reg_exp,
+                                    bug_system.validate_reg_exp))
```

Comment:
Why not json.dumps(bug_system.serialize()) ?

Reply:
Not recreating the whole model, just saving & restoring the 3 fields since the object wasn't deleted. Wouldn't restoring it also need to recreate the whole object and might conflict with the already existing one?

Reply:
Give it a try like shown in the original restore snippet and you will see. Should not have any issues. Furthermore `update_fields` parameter to `.save()` will instuct the ORM to prepare an update query only for the specified fields and not all of the rest even if they are present in the values dict.

Reply:
CI error in both `test-for-missing-migrations` & `test-migrations-rollback` :

`AttributeError: 'BugSystem' object has no attribute 'serialize'`

Reply:
Make the Bug model inherit from `TCMSActionModel` instead of the current base classes and that will resolve the missing method. Serialization is unfortunately another field which is desperate for refactoring but will work with the current state for now.

Reply:
Should I leave it as is then? Seems like the non-json approach works too

Reply:
The non-json approach is not elegant and prone to failure. Also it needs to be changed in all of the other places as well so it's better to make it actually work. Cherry-pick my last 2 commits from https://github.com/kiwitcms/Kiwi/pull/1846 and keep the commit log explanations in the first one. 

Also notice there are some print statements and how they appear in the logs. 

Note: I haven't disabled warnings for bandit to make the changes more readable while experimenting with them. 


The rest of the migrations need to be updated accordingly wrt serialization.

Reply:
Made changes.
> The rest of the migrations need to be updated accordingly wrt serialization.

Which other ones? It looked like it was just the BugSystem model in `/testcases/migrations/0011_trim_bugsystem_fields.py` that had problems wrt serialization.

Reply:
Read what the commit log for the last commit says. There's a reason you don't see other failures and it is explained in the commit message.

Reply:
Didn't quite understand but I think I get it now. Pushed changes for the other two model dumps.